src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/CloudStackCloudUtils.java

```diff
@@ -49,61 +49,77 @@ public static String doRequest(@NotNull CloudStackHttpClient cloudStackHttpClien
         }
     }
 
-    // TODO (chico) - move to the CloudstackCloudUtil
+    /**
+     * Wait and process the Cloudstack asynchronous response in its asynchronous life cycle.
+     * @throws FogbowException
+     */
     public static String waitForResult(@NotNull CloudStackHttpClient client,
                                        String cloudStackUrl,
                                        String jobId,
                                        @NotNull CloudStackUser cloudStackUser)
             throws FogbowException {
-        CloudStackQueryAsyncJobResponse queryAsyncJobResult = getAsyncJobResponse(client,
-                cloudStackUrl, jobId, cloudStackUser);
-
-        if (queryAsyncJobResult.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
-            for (int i = 0; i < MAX_TRIES; i++) {
-                queryAsyncJobResult = getAsyncJobResponse(client, cloudStackUrl, jobId, cloudStackUser);
-                if (queryAsyncJobResult.getJobStatus() != CloudStackQueryJobResult.PROCESSING) {
-                    return processJobResult(queryAsyncJobResult, jobId);
-                }
-                try {
-                    Thread.sleep(ONE_SECOND_IN_MILIS);
-                } catch (InterruptedException e) {
-                    throw new FogbowException();
-                }
+
+        int countTries = 0;
+        CloudStackQueryAsyncJobResponse response = getAsyncJobResponse(
+                client, cloudStackUrl, jobId, cloudStackUser);
+        while(response.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
+            if (countTries >= MAX_TRIES) {
+                throw new TimeoutCloudstackAsync(String.format(Messages.Exception.JOB_TIMEOUT, jobId));
```

Comment:
Is this exception created by you? If so I suggest to change it. CloudStackTimeoutException seems better for me

Reply:
I agree! I forgot the "Exception" in the class name...

Reply:
I will make this change in the last PR(five), right?

Reply:
Done